src/utils.jl

```diff
@@ -192,7 +192,18 @@ end
 function read_json_str(json)
     # length check is to ensure that isfile doesn't thrown an error
     # see issue for details https://github.com/JuliaLang/julia/issues/39774
-    !(json isa VectorString) && sizeof(json) < 255 && isfile(json) ?
-          VectorString(Mmap.mmap(json)) : 
-          json
+    isfilename(json) && isfile(json) ?
+        VectorString(Mmap.mmap(json)) :
+        json
+end
+
+isfilename(filename) = try
+    stat(filename)
+    true
+catch e
+    if isa(e, Base.IOError)
+        e.code != -36   # -36 is ENAMETOOLONG; other IO errors indicate valid filename
```

Comment:
is this code consistent across OS platforms? can we add a test to catch the new code paths here?

Reply:
I suspect the original code might work as intended on Windows, and this branch wouldn't ever be triggered. Still, yeah I'll try to write a test.